src/App.tsx

```diff
@@ -1,24 +1,247 @@
+import { FilterBy } from './types/FilteredBy';
+import { TodoFilter } from './components/TodoFilter';
+import { Form } from './components/Form';
+import { ErrorMessage } from './components/ErrorMessage';
+import { setSingleOrPluralWordByCount } from './HelpersFunctions';
 
-const USER_ID = 0;
+const USER_ID = 6894;
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+function filterTodosByCompleted(todos: Todo[], filterBy: FilterBy): Todo[] {
+  switch (filterBy) {
+    case FilterBy.ACTIVE:
+      return todos.filter(todo => !todo.completed);
+
+    case FilterBy.COMPLETED:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
   }
+}
+
+export interface TodoItem {
+  todo: Todo,
+  isLoading: boolean,
+}
+
+export const App: FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterType, setFilterType] = useState<FilterBy>(FilterBy.ALL);
+  const [tempTodos, setTempTodos] = useState<Todo | null>(null);
+  const [activeInput, setActiveInput] = useState(true);
+  const [isQuery, setIsQuery] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
+
+  const addError = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const loadAllTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setAllTodos(todos);
+      setIsQuery(true);
+    } catch {
+      addError('Unable to update a todo');
+    } finally {
+      setTempTodos(null);
+      setActiveInput(true);
+    }
+  };
+
+  const handleRemoveTodo = useCallback(
+    async (id: number) => {
+      try {
+        setDeletedTodoIds(prev => [...prev, id]);
+        await removeTodo(id)
+          .then(() => {
+            const todosWithoutDeleted = allTodos.filter((todo) => (
+              todo.id !== id
+            ));
+
+            setAllTodos(todosWithoutDeleted);
+          });
+      } catch {
+        setErrorMessage('Unable to delete a todo');
+      } finally {
+        setDeletedTodoIds([]);
+      }
+    }, [allTodos],
+  );
+
+  useEffect(() => {
+    loadAllTodos();
+  }, []);
+
+  const handleRemoveAllComplitedTodo = useCallback(async () => {
+    const completedTodos = allTodos.filter(({ completed }) => completed);
+
+    completedTodos.forEach(async todo => {
+      try {
+        setDeletedTodoIds(prev => [...prev, todo.id]);
+        await removeTodo(todo.id)
+          .then(() => {
+            const todosWithoutDeleted = allTodos.filter(({ completed }) => (
+              !completed
+            ));
+
+            setAllTodos(todosWithoutDeleted);
+          });
+      } catch {
+        setErrorMessage('Unable to delete a todo');
+      } finally {
+        setDeletedTodoIds([]);
```

Comment:
this will reset all loading todos when at least one of them will be removed

Reply:
if you want to disable all loading todos at one time 

1.  pass all ids to array before running async function for each todo
2. replace `forEach` with map
3. use Promise.all for this map
4. await this promise
5. then set empty array to loading todo ids